crates/rpc/rpc-eth-api/src/helpers/mod.rs

```diff
@@ -45,3 +45,21 @@ pub trait TraceExt:
 }
 
 impl<T> TraceExt for T where T: LoadTransaction + LoadBlock + LoadPendingBlock + Trace + Call {}
+
+/// Helper trait to unify all `eth` rpc server building block traits, for simplicity.
+pub trait FullEthApiServerGateway:
```

Comment:
can we drop the `ServerGateway` here?

Reply:
> An API gateway is a data-plane entry point for API calls that represent client requests to target applications and services.

the descriptive name is very helpful. would you be ok with `FullEthApiGateway`?

Reply:
https://www.f5.com/glossary/api-gateway